src/data/move-attrs/random-moveset-move-attr.ts

```diff
@@ -1,67 +1,154 @@
-import type { BattlerIndex } from "#enums/battler-index";
-import { MoveFlags } from "#enums/move-flags";
+import { allMoves } from "#app/data/all-moves";
+import { type Move } from "#app/data/move";
+import { CallMoveAttr } from "#app/data/move-attrs/call-move-attr";
+import type { MoveConditionFunc } from "#app/data/move-conditions";
 import type { Pokemon } from "#app/field/pokemon";
 import { globalScene } from "#app/global-scene";
-import { MovePhase } from "#app/phases/move-phase";
-import { type Move, getMoveTargets } from "#app/data/move";
-import { OverrideMoveEffectAttr } from "#app/data/move-attrs/override-move-effect-attr";
+import type { BooleanHolder } from "#app/utils";
+import { MoveId } from "#enums/move-id";
 
 /**
- * Attribute to invoke a random move from the user or enemy's moveset
- * and use it against a random legal target.
- * Used by {@link https://bulbapedia.bulbagarden.net/wiki/Sleep_Talk_(move) | Sleep Talk}
- * and {@link https://bulbapedia.bulbagarden.net/wiki/Assist_(move) | Assist}.
- * @extends OverrideMoveEffectAttr
+ * Attribute used to call a random move in the user or party's moveset.
+ * Used for {@linkcode MoveId.ASSIST} and {@linkcode MoveId.SLEEP_TALK}
+ *
+ * Fails if the user has no callable moves.
+ * @extends CallMoveAttr
+ * @see {@linkcode getCondition} for move selection
  */
-export class RandomMovesetMoveAttr extends OverrideMoveEffectAttr {
-  private enemyMoveset: boolean | null;
+export class RandomMovesetMoveAttr extends CallMoveAttr {
+  private includeParty: boolean;
```

Comment:
Q: Does this ever change after int? Otherwise => readonly

```suggestion
  private readonly includeParty: boolean;
```